Source/NSCollectionView.m

```diff
@@ -603,35 +885,68 @@ - (id) initWithCoder: (NSCoder *)aCoder
   
   if (self)
     {
-      _itemSize = NSMakeSize(0, 0);
-      _tileWidth = -1.0;
-          
+      [self _initDefaults];
       if ([aCoder allowsKeyedCoding])
         {
-          _minItemSize = [aCoder decodeSizeForKey: NSCollectionViewMinItemSizeKey];
-          _maxItemSize = [aCoder decodeSizeForKey: NSCollectionViewMaxItemSizeKey];
+          if ([aCoder containsValueForKey: NSCollectionViewMinItemSizeKey])
+            {
+              _minItemSize = [aCoder decodeSizeForKey: NSCollectionViewMinItemSizeKey];
+            }
+          
+          if ([aCoder containsValueForKey: NSCollectionViewMaxItemSizeKey])
+            {
+              _maxItemSize = [aCoder decodeSizeForKey: NSCollectionViewMaxItemSizeKey];
+            }
+          
+          if ([aCoder containsValueForKey: NSCollectionViewMaxNumberOfRowsKey])
+            {
+              _maxNumberOfRows = [aCoder decodeInt64ForKey: NSCollectionViewMaxNumberOfRowsKey];
+            }
           
-          _maxNumberOfRows = [aCoder decodeInt64ForKey: NSCollectionViewMaxNumberOfRowsKey];
-          _maxNumberOfColumns = [aCoder decodeInt64ForKey: NSCollectionViewMaxNumberOfColumnsKey];
+          if ([aCoder containsValueForKey: NSCollectionViewMaxNumberOfColumnsKey])
+            {
+              _maxNumberOfColumns = [aCoder decodeInt64ForKey: NSCollectionViewMaxNumberOfColumnsKey];
+            }
           
           //_verticalMargin = [aCoder decodeFloatForKey: NSCollectionViewVerticalMarginKey];
           
-          _isSelectable = [aCoder decodeBoolForKey: NSCollectionViewSelectableKey];
-          _allowsMultipleSelection = [aCoder decodeBoolForKey: NSCollectionViewAllowsMultipleSelectionKey];
+          if ([aCoder containsValueForKey: NSCollectionViewSelectableKey])
+            {
+              _isSelectable = [aCoder decodeBoolForKey: NSCollectionViewSelectableKey];
+            }
+          
+          if ([aCoder containsValueForKey: NSCollectionViewAllowsMultipleSelectionKey])
+            {
+              _allowsMultipleSelection = [aCoder decodeBoolForKey: NSCollectionViewAllowsMultipleSelectionKey];
+            }
+          
+          if ([aCoder containsValueForKey: NSCollectionViewBackgroundColorsKey])
```

Comment:
Not sure why this is working as you did not override the containsValueForKey: method for this key.

Reply:
I don't believe the current test code exercises this.